voluseg/_steps/step0.py

```diff
@@ -1,182 +1,256 @@
 
     # check plane padding
-    if (not parameters['registration']) and not ((parameters['planes_pad'] == 0)):
-        raise Exception('\'planes_pad\' must be 0 if \'registration\' is None.')
+    if (not parameters["registration"]) and not ((parameters["planes_pad"] == 0)):
+        raise Exception("'planes_pad' must be 0 if 'registration' is None.")
 
     # convert dir_input into a list to account for multiple directories
-    input_dirs = [os.path.normpath(h) for h in dir_input.split(';')]
+    input_dirs = [os.path.normpath(h) for h in dir_input.split(";")]
     if len(input_dirs) == 1:
         prefix_dirs = [None]
     else:
         prefix_dirs = [os.path.basename(h) for h in input_dirs]
         if len(prefix_dirs) != len(set(prefix_dirs)):
-            raise Exception('the names of last directories in \'dir_input\' must be unique.')
+            raise Exception(
+                "the names of last directories in 'dir_input' must be unique."
+            )
         if prefix_dirs != sorted(prefix_dirs):
-            raise Exception('the names of last directories in \'dir_input\' must be sorted.')
+            raise Exception(
+                "the names of last directories in 'dir_input' must be sorted."
+            )
 
     volume_fullnames_input = []
     volume_names = []
-    for dir_input_h, dir_prefix_h in zip(input_dirs, prefix_dirs):
-        # get volume extension, volume names and number of segmentation timepoints
-        file_names = [i.split('.', 1) for i in os.listdir(dir_input_h) if '.' in i]
-        file_exts, counts = np.unique(list(zip(*file_names))[1], return_counts=True)
-        ext = '.'+file_exts[np.argmax(counts)]
-        volume_names_input_h = sorted([i for i, j in file_names if '.'+j == ext])
-        volume_fullnames_input_h = [os.path.join(dir_input_h, i) for i in volume_names_input_h]
-
-        # adjust parameters for packed planes data
-        if parameters['planes_packed']:
-            parameters['res_z'] = parameters['diam_cell']
-
-            def get_plane_names(tuple_fullname_volume_input):
-                fullname_volume_input = tuple_fullname_volume_input[1]
-                lp = len(load_volume(fullname_volume_input+ext))
-                return [get_volume_name(fullname_volume_input, dir_prefix_h, pi) for pi in range(lp)]
-
-            volume_names_h = evenly_parallelize(volume_fullnames_input_h).map(get_plane_names).collect()
-            volume_names_h = [pi for ni in volume_names_h for pi in ni]
-        else:
-            volume_names_h = [get_volume_name(i, dir_prefix_h) for i in volume_names_input_h]
-
-        # grow volume-name lists
-        volume_fullnames_input += volume_fullnames_input_h
-        volume_names += volume_names_h
-
-    volume_fullnames_input = np.array(volume_fullnames_input)
-    volume_names = np.array(volume_names)
-    lt = len(volume_names)
+    if parameters.get("nwb_input_local_path", None):
```

Comment:
It seems redundant to add include `nwb_input_local_path` and `nwb_input_acquisition_name` variables since this presumably can just be the name of `dir_input`.

The idea would be that voluseg will automatically detect the 4D nwb file in the `input_dir` and set the extension to that file.

Alternatively, users can just specify `ext` as a parameter to force the detection of files with specific extensions.

Reply:
@mikarubi 
One nwb file typically contains the whole data for an experimental session for a single subject. Would there be any cases where a single voluseg pipeline should be used to simultaneously process the data from multiple sessions?

The argument `nwb_input_acquisition_name` serves to identify the correct TwoPhotonSeries object within the nwb file, which at least in principle could store more than one TwoPhotonSeries. We can certainly have an automatic detection of this type of objects inside the nwb file, though!

What about we make this argument optional? In the case of of a single TwoPhotonSeries, the pipeline progresses automatically using that one, if there are multiple TwoPhotonSeries the pipeline will break with an Error message requesting the user to provide `nwb_input_acquisition_name`.

Nwb also allows for MotionCorrection data, which goes as a different object type, stored in a different location. TwoPhotonSeries objects are stored in the `acquisition` path, while MotionCorrection objects are stored inside a processing submodule. Here again, we can iterate over all objects within the nwb file and detect existing MotionCorrection objects, but we should probably have an argument that allows the user to tell explicitly which object they want to process.

Reference: https://pynwb.readthedocs.io/en/stable/tutorials/domain/ophys.html#two-photon-series

Reply:
Yes, my suggestion is to keep things simple -- i.e. automatically fetch the name (see example below) and throw error if there are more than two sessions (unless it is common to store the data in this way). I do it already with hdf5:
 https://github.com/mikarubi/voluseg/blob/7b780e5c3d32593907071e9a076bb64b0ea6046b/voluseg/_tools/load_volume.py#L33

I didn't realize that there is a separate location for motion corrected data. It seems simple to just add another option to the registration: "nwb_motion_correction". This will signal that voluseg will fetch the Motion Corrected data from NWB, and will disable internal ANTs motion correction. This should not interfere with functionality because registration (i.e. motion correction) should always be disabled when the user is already planning to use previously corrected data.